### PR TITLE
Fix for meta character being sent in request body during save

### PR DESCRIPTION
**Pull request by lilHermit**
proposed fix for #105

### CODE REVIEW

File: src/Type.php [hunk 1]
```diff
@@ -469,8 +469,10 @@ public function save(EntityInterface $entity, $options = [])
         $type = $this->connection()->getIndex()->getType($this->name());
         $id = $entity->id ?: null;
 
-        $data = $entity->toArray();
-        unset($data[$id]);
+        $keys = array_keys($entity->toArray());
+        $metaKeys = ['id', '_version'];
+        $data = $entity->extract(array_diff($keys, $metaKeys));
```
Comment (lorenzo) [inline src/Type.php]: I think you can just call `unset($data[$id], $data['_version'])` instead of this

Comment (lilHermit) [inline src/Type.php]: Yeah just thought this was more future proofed.

Also it would need to be `unset($data['id'], $data['_version']`

Comment (lorenzo) [inline src/Type.php]: Right... since it is just 2 fields for now, let's just use `unset` if we need more fields we can then use a more generic approach.

Comment (lilHermit) [inline src/Type.php]: Ok what is the best approach from here? Shall I revert this commit (locally), make a new one and then create another pull request for that new commit?

Comment (lorenzo) [inline src/Type.php]: just push a new commit

**Comment by markstory** [comment]
Could we get a test for this? If you don't know how to do the test I'm happy to help.

**Comment by lilHermit** [comment]
@markstory Yeah I'm happy to do a test but will need some guidance.

**Comment by markstory** [comment]
@lilHermit You could add a test that double saves an entity. That triggered the problem before didn't it?

**Comment by lilHermit** [comment]
Double Save test added

**Comment by lorenzo** [comment]
@markstory was that the test case you were expecting to see?

**Comment by markstory** [comment]
@lorenzo Yeah.

**Comment by lorenzo** [comment]
Thanks @lilHermit

**Pull request merged**